lingua_franca/lang/common_data_pl.py

```diff
@@ -355,23 +355,164 @@
     'grudnia': 'December',
 }
 
+# https://odmiana.net/ 
+# Terms of service state clearly (22.09.2020):
+# Definicje:
+# (...)
+# Użytkownik - osoba fizyczna.
+# (...)
+# Ogólne warunki:
+# (...)
+# Portal przeznaczony jest dla wszystkich Użytkowników Internetu.
+# Dostęp do portalu a także korzystanie z niego jest całkowicie bezpłatne dla wszystkich Użytkowników.
+# Which means:
+# Definitions:
+# (...)
+# User - person.
+# (...)
+# General terms:
+# (...)
+# This site is intended for all Internet Users.
+# Accessing the site and using it is completely free of charge for all Users.
```

Comment:
Hi Tomasz,

Thanks for all the work you've been doing.

I wanted to check about this comment block. Is it saying that these translations were sourced from that website?

Unfortunately the permission to access a site freely does not necessarily imply that the contents can be freely re-used elsewhere. 

The bad news is that unless there is an explicit and compatible license, or we receive explicit permission and release from the site owners, we cannot accept this contribution. This means means that we need to source those translations directly from native speakers or similarly licensed projects (though I don't think we've really done that yet).

A side benefit of doing this is that everything contributed is released under the permissive Apache 2.0 license. As such anyone wanting to use translations from it can do so in basically whatever way they like. So it's slower getting started, however we are creating a resource not just for Mycroft, it is then available for anyone working with language processing.

Reply:
Yes, these cases (well, most of them) are generated by the online service odmiana.net. It's not like one can patent a language, but I am very concerned with my work being correct in legal terms, so I will contact the site maintainer to ask for explicit use of this work and it's usage in this project.

After reading their TOS I assumed that I can use it's output. I'm sorry if that was inappropriate. Please give me some time to get a response from the site Administrator.

Reply:
Yeah the whole topic of data sovereignty over language is a pretty interesting area, particularly for "minority" languages. It's a big issue in Australia and one of the reasons I believe so passionately in the need for open source voice tech. 

There's no harm in trying the site owners, however I'm fairly doubtful that they'll be open to it given their business model appears to be ad-based. If they did agree we'd also need some way to verify their licensing of that data or the release of it into the public domain. Alternatively they could sign our corporate CLA. 

It's kind of a messy process which is why we generally steer clear of it, but I'll leave it with you.

Reply:
how do you license words? i mean, i can put up a website with all numbers in portuguese and that makes it so you can't use it? languages are public domain essentially.... it's crazy that you need to ask permission to use the names of numbers?

Reply:
I have already contacted the Administrator. I am waiting for a response. I am optimistic, because we're not asking for an API access, just already generated weekdays grammatical cases.

Reply:
@JarbasAI To be honest... I do not own a paper dictionary :-).

Reply:
my argument is that reaching out to them is nothing more than a courtesy, language is not patented or licensed

Reply:
Yes, I am a Polish speaker. In terms of opinions I absolutely agree with you. Then again, as a contributor I must respect the rules of the project or my work may not be accepted. And I want it to be accepted.

Reply:
I fully agree that you shouldn't be able to copyright a language, however in effect that's exactly what happens. Here's the [Macquarie Dictionary license](https://web.macquariedictionary.com.au/site/licence). Of particular note:
> Except as expressly permitted in Clause 2.1, the Licensee warrants that it will not, nor will it license or permit others to, directly or indirectly, without Macquarie's prior written consent:
...
(e) use the Licensed Material to create any derivative work, product or service, or merge the Licensed Material with any other product, database, or service;

Even more interesting is a right they do grant:
> (d) create a hypertext link to any part of the Licensed Material provided that no person other than the Licensee may use such hypertext link.

So technically linking to any part of their site that contains "Licensed Material" would violate the terms of this license agreement...

I'm sure their argument would be that they are't licensing the language itself, but their collation and interpretation of it. 

It's an enormous legal grey area, and you can quickly take it to comical extremes... If I use the Macquarie Dictionary to learn English and then create anything using my newly acquired language skills, is that a derivative work? Seems stupid until someone gets sued for it... (actually it would still be stupid).

At the end of the day, we can't knowingly accept potentially copyrighted material no matter how fundamental it is. The same goes for other projects. If you contribute to OpenStreetMap you can't just open a Google Maps window and start copying across street names. The street names themselves aren't owned by Google, but the digital collection of them accessible through Google Maps is.

Reply:
I could rant about this a lot... maybe I should write a blog post :D

Reply:
OK, so I didn't get a response from the odmiana.net Administrator, BUT I got in touch with a friend who owns a paper dictionary. Then after looking up week days and word variation rules we were able to recreate this result. We used the "Słownik współczesnego języka polskiego" (The Modern Polish dictionary), published in 1996 by "Wilga" publishing house (ISBN 83-7156-068-0). Pages 20-28 describe the nouns and adjectives (yes, we have words like "Monday-ish"!) variation rules, as annotated for each week day. It took us some time, but fortunately we came up with the same results. How should I abandon the unneeded service disclaimer?

Reply:
I'm no lawyer, but I'm inclined to agree, because you aren't republishing the dictionary's definitions. You're setting up data *according* to the relevant definitions in a reputable dictionary.

I'll wait for Gez to opine.

As to relicensing, if the copyright situation is sound, no need to do anything further for LF's purposes. Your contribution is covered by the CLA, becomes part of LF, and LF is currently Apache-licensed. If you want to distribute your bit separately, you can license it however you want.

The importing bit here, afaik, is that the code portion was not lifted from that website *or* from the dictionary. Rather, those were references in the compilation of this data.

Reply:
The dictionary contains only the word and annotations that briefly points to particular rules on how to create it's variation. Then using rules I create derivative work by generating all valid nouns and adjectives. In other words, I can find there *only* the main noun: "poniedziałek", but not "poniedziałku", "poniedziałkowi", "poniedziałkom", etc.

Reply:
@krisgesling has the final word, but i think you just need to remove the code comment about odmiana and it's good to go

this whole situation is just sad, now that mycroft is legally covered lets pretend this nonsense never happened

Reply:
Done

Reply:
Hey, thanks for updating this. It's inspired a blog post as it is a stupid situation but unfortunately one we have to deal with.

Reply:
I'd be happy to read it :).